libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/TilesetDescriptorFactory.kt

```diff
@@ -0,0 +1,68 @@
+package com.mapbox.navigation.core.navigator
+
+import com.mapbox.common.TilesetDescriptor
+import com.mapbox.navigation.navigator.internal.MapboxNativeNavigator
+import com.mapbox.navigator.CacheHandle
+
+private typealias NativeTilesetDescriptorFactory = com.mapbox.navigator.TilesetDescriptorFactory
+
+/**
+ * A factory to build navigation [TilesetDescriptor]
+ */
+class TilesetDescriptorFactory internal constructor(
+    navigator: MapboxNativeNavigator
+) {
+    private val cache: CacheHandle = navigator.cache!!
```

Comment:
Is this always non-null?

Reply:
yes, must be non-null

Reply:
Why is the type nullable in this case?

Reply:
all of the objects are nullable in `MapboxNativeNavigatorImpl`:
```kotlin
private var navigator: Navigator? = null
    private var nativeRouter: Router? = null
    private var historyRecorderHandle: HistoryRecorderHandle? = null
    private var route: DirectionsRoute? = null
    private var routeBufferGeoJson: Geometry? = null
    override var graphAccessor: GraphAccessor? = null
    override var openLRDecoder: OpenLRDecoder? = null
    override var roadObjectsStore: RoadObjectsStore? = null
    override var cache: CacheHandle? = null
    private var logger: Logger? = null
```
and are inited in `override fun create()` and it must be called before any call to `MapboxNativeNavigatorImpl`

we also have a lot of calls with ` navigator!!`
so I think it's ok to have `navigator.cache!!`